core/src/test/resources/org/apache/calcite/test/SqlToRelConverterTest.xml

```diff
@@ -3813,11 +3813,10 @@ LogicalProject(DEPTNO=[$9], SAL=[$7])
     LogicalTableScan(table=[[CATALOG, SALES, DEPT]])
     LogicalJoin(condition=[true], joinType=[left])
       LogicalTableScan(table=[[CATALOG, SALES, EMP]])
-      LogicalAggregate(group=[{}], agg#0=[SINGLE_VALUE($0)])
-        LogicalProject(EXPR$0=[>($0, 0)])
-          LogicalAggregate(group=[{}], agg#0=[AVG($0)])
-            LogicalProject(SAL=[$5])
-              LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+      LogicalProject(EXPR$0=[>($0, 0)])
+        LogicalAggregate(group=[{}], agg#0=[AVG($0)])
```

Comment:
There's only one row so $0 is unique, and so `Blackboard.reRegister` is satisfied:
```
          relBuilder.aggregate(relBuilder.groupKey(),
              relBuilder.aggregateCall(SqlStdOperatorTable.SINGLE_VALUE,
                  relBuilder.field(0)));
```
doesn't need an additional aggregation.